iterators/1-each.rb

```diff
@@ -0,0 +1,144 @@
+rainbow = %w[red orange yellow green gray indogo violet]
+
+# use cicle for
+for color in rainbow
+  puts color
+end
+
+# red
+# orange
+# yellow
+# green
+# gray
+# indogo
+# violet
+
+puts
+# use each
+rainbow.each { |color| puts color }
+
+# times
+5.times {|t| puts t}
+
+# 0
+# 1
+# 2
+# 3
+# 4
+
+puts
+# upto and downto
+5.upto(10) {|t| puts t}
+
+15.downto(10) {|t| puts t}
+
+# Iterator            Synonym
+# each
+# each_with_index
+# map                 collect
+# select              find_all
+# reject              delete_if
+# reduce              inject
+# each_with_object
+
+puts
+# each, no mutation
+arr = [1, 2, 3, 4, 5]
+
+res = arr.each { |x| puts x ** 2 }
+p res
+1
+4
+9
+16
+25
+[1, 2, 3, 4, 5] # returns arr
+
+puts
+# each_with_index
+rainbow = %w[red orange yellow green gray indogo violet]
+rainbow.each_with_index { |color, ind| puts "#{ind}: #{color}" }
+
```

Comment:
> In the same case when bypassing an array of strings or characters, we often need to number the elements. For this, it is convenient to use the `each_with_index` iterator. 

_This iterator block takes two parameters: 

- [ ] the first is the element of the collection, 
- [ ] and the second is the current number